.github/workflows/conformance-aks.yaml

```diff
@@ -427,10 +427,10 @@ jobs:
   merge-upload-and-status:
     name: Merge Upload and Status
     if: ${{ always() && !inputs.is-workflow-call }}
-    needs: installation-and-connectivity
+    needs: [generate-matrix, wait-for-images, installation-and-connectivity]
     uses: ./.github/workflows/common-post-jobs.yaml
     secrets: inherit
     with:
       context-ref: ${{ inputs.context-ref || github.sha }}
       sha: ${{ inputs.SHA || github.sha }}
-      result: ${{ needs.installation-and-connectivity.result }}
+      result: ${{ needs.wait-for-images.result == 'success' && needs.generate-matrix.result == 'success' && needs.installation-and-connectivity.result == 'success' && 'success' || 'failure' }}
```

Comment:
This change defeats the original intention of https://github.com/cilium/cilium/commit/54796b085eb2f8c695db74cef95521e732effacf, as now the workflow would be always marked as failed if the generated matrix is empty.

Reply:
> What if we remove the generate-matrix from it

That has the drawback of marking the workflow as successful if the matrix generation fails for whatever reason, which IIRC happened fairly recently.

> I assume you meant this comment to be applied for all cloud providers?

Yep, all the workflows in which the generated matrix can be legitimately empty.